lua/typescript-tools/api.lua

```diff
@@ -27,4 +27,39 @@ function M.request_diagnostics(callback)
   }, callback)
 end
 
+--- Returns an |lsp-handler| that filters TypeScript diagnostics with the given codes.
+--- <pre>lua
+--- local api = require('typescript-tools.api')
+--- require('typescript-tools').setup {
+---   handlers = {
+---     -- Ignore 'This may be converted to an async function' diagnostics.
+---     ['textDocument/publishDiagnostics'] = api.filter_diagnostics { 80006 }
+---   }
+--- }
+--- </pre>
+---
+---@param codes integer[]
+function M.filter_diagnostics(codes)
+  return function(err, res, ctx, config)
+    local filtered = {}
+    for _, diag in ipairs(res.diagnostics) do
+      if diag.source == "tsserver" then
+        local code_filtered = false
+        for _, code in ipairs(codes) do
+          if diag.code == code then
+            code_filtered = true
+            break
+          end
+        end
```

Comment:
You can use `vim.tbl_add_reverse_lookup` in enclosing scope to easily check is code blacklisted using `codes[diag.code] ~= nil`. I know It's not that much overhead in having this few extra iterations, but I think this way it will be easier to read.